src/components/MultiGestureCanvas/index.tsx

```diff
@@ -0,0 +1,269 @@
+    const offsetX = useSharedValue(0);
+    const offsetY = useSharedValue(0);
+
+    /**
+     * Stops any currently running decay animation from panning
+     */
+    const stopAnimation = useWorkletCallback(() => {
+        cancelAnimation(offsetX);
+        cancelAnimation(offsetY);
+    });
+
+    /**
+     * Resets the canvas to the initial state and animates back smoothly
+     */
+    const reset = useWorkletCallback((animated: boolean, callbackProp?: () => void) => {
+        const callback = callbackProp ?? (() => {});
+
+        pinchScale.value = 1;
+
+        stopAnimation();
+
+        pinchScale.value = 1;
+
+        if (animated) {
+            offsetX.value = withSpring(0, SPRING_CONFIG);
+            offsetY.value = withSpring(0, SPRING_CONFIG);
+            panTranslateX.value = withSpring(0, SPRING_CONFIG);
+            panTranslateY.value = withSpring(0, SPRING_CONFIG);
+            pinchTranslateX.value = withSpring(0, SPRING_CONFIG);
+            pinchTranslateY.value = withSpring(0, SPRING_CONFIG);
+            zoomScale.value = withSpring(1, SPRING_CONFIG, callback);
+            return;
+        }
+
+        offsetX.value = 0;
+        offsetY.value = 0;
+        panTranslateX.value = 0;
+        panTranslateY.value = 0;
+        pinchTranslateX.value = 0;
+        pinchTranslateY.value = 0;
+        zoomScale.value = 1;
+
+        callback();
+    });
+
+    const {singleTapGesture: basicSingleTapGesture, doubleTapGesture} = useTapGestures({
+        canvasSize,
+        contentSize,
+        minContentScale,
+        maxContentScale,
+        offsetX,
+        offsetY,
+        pinchScale,
+        zoomScale,
+        reset,
+        stopAnimation,
+        onScaleChanged,
+        onTap,
+    });
+    const singleTapGesture = basicSingleTapGesture.requireExternalGestureToFail(doubleTapGesture, panGestureRef);
+
+    const panGestureSimultaneousList = useMemo(
+        () => (pagerRef === undefined ? [singleTapGesture, doubleTapGesture] : [pagerRef as unknown as Exclude<GestureRef, number>, singleTapGesture, doubleTapGesture]),
+        [doubleTapGesture, pagerRef, singleTapGesture],
+    );
+
+    const panGesture = usePanGesture({
+        canvasSize,
+        contentSize,
+        zoomScale,
+        totalScale,
+        offsetX,
+        offsetY,
+        panTranslateX,
+        panTranslateY,
+        isPagerSwiping,
+        stopAnimation,
+    })
+        .simultaneousWithExternalGesture(...panGestureSimultaneousList)
+        .withRef(panGestureRef);
+
+    const pinchGesture = usePinchGesture({
+        canvasSize,
+        zoomScale,
+        zoomRange,
+        offsetX,
+        offsetY,
+        pinchTranslateX,
+        pinchTranslateY,
+        pinchScale,
+        isPagerSwiping,
+        stopAnimation,
+        onScaleChanged,
+    }).simultaneousWithExternalGesture(panGesture, singleTapGesture, doubleTapGesture);
```

Comment:
```suggestion
    }).simultaneousWithExternalGesture(panGesture, doubleTapGesture);
```

singleTapGesture was not being passed

Reply:
idk why, but in my code `singleTapGesture` is there. resolving nevertheless